Document AnnData output

allyhawkins opened the pull request:
Closes #367 

This PR updates the documentation in this repo to reflect that AnnData objects are now output by the workflow. 

- I updated the main README to note that output is both SCE objects and AnnData. 
- I modified the outputs section in the external instructions to mention that files produced will be both the rds and the hdf5 files. I also updated the expected file output structure and grouped by file type to make things easier to read. 
- I did not include `_adt.hdf5` files in the output structure, but I do mention them in the text. Do we think that's sufficient?
- While working on this I made some changes in the docs to reflect our change to use ADT when talking about CITE-seq.

sjspielman reviewed (commented):
This looks fine to me overall, just a couple very minor phrasing comments. 

My main comment is about the multiplexed libraries and AnnData, since I know that is planned but on hold, right? Maybe we should open an issue to circle back to docs once that is finalized so we can add in the right docs for however that pans out?

allyhawkins commented:
Thanks for the suggestions. I incorporated the ones you added and then also added cell hashing to the main readme. I think that just hadn't been updated since we added it in so good catch! 
I also added a note to the AnnData output description that no AnnData objects are provided for multiplexed libraries. 
I'm not sure if we really are going to start exporting cell hashed libraries as AnnData just yet, so I'm going to just add a note to the existing issue to make sure we update documentation accordingly if we do make that change.

sjspielman reviewed (approved):
🚀
